aws-rds-dbinstance/src/main/java/software/amazon/rds/dbinstance/CreateHandler.java

```diff
@@ -0,0 +1,125 @@
+package software.amazon.rds.dbinstance;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Optional;
+
+import com.amazonaws.util.StringUtils;
+import software.amazon.awssdk.services.rds.RdsClient;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+import software.amazon.cloudformation.resource.IdentifierUtils;
+
+public class CreateHandler extends BaseHandlerStd {
+
+    protected ProgressEvent<ResourceModel, CallbackContext> handleRequest(
+            final AmazonWebServicesClientProxy proxy,
+            final ResourceHandlerRequest<ResourceModel> request,
+            final CallbackContext callbackContext,
+            final ProxyClient<RdsClient> proxyClient,
+            final Logger logger
+    ) {
+        return ProgressEvent.progress(request.getDesiredResourceState(), callbackContext)
+                .then(progress -> {
+                    final ResourceModel model = request.getDesiredResourceState();
+                    if (StringUtils.isNullOrEmpty(model.getDBInstanceIdentifier())) {
+                        model.setDBInstanceIdentifier(IdentifierUtils.generateResourceIdentifier(
+                                request.getStackId(),
+                                Optional.ofNullable(request.getLogicalResourceIdentifier()).orElse("dbinstance"),
+                                request.getClientRequestToken(),
+                                DB_INSTANCE_ID_MAX_LENGTH
+                        ).toLowerCase());
+                    }
+                    final Map<String, String> tags = mergeMaps(Arrays.asList(
+                            request.getSystemTags(),
+                            request.getDesiredResourceTags()
+                    ));
+                    model.setTags(Translator.translateTagsFromRequest(tags));
+                    return ProgressEvent.progress(model, progress.getCallbackContext());
+                })
+                .then(progress -> {
+                    if (isReadReplica(progress.getResourceModel())) {
+                        return initiateCreateDbInstanceReadReplicaRequest(proxy, proxyClient, progress);
+                    } else if (isRestoreFromSnapshot(progress.getResourceModel())) {
+                        return initiateRestoreDbInstanceFromSnapshotRequest(proxy, proxyClient, progress);
+                    }
+                    return initiateCreateDbInstanceRequest(proxy, proxyClient, progress);
+                })
+                .then(progress -> waitForDbInstanceAvailableStatus(proxy, proxyClient, progress))
+                .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
```

Comment:
Do we need to reboot the instance if it has a parameter group name? Also, are we calling `addRoleToDBInstance` from the create handler somewhere?

Reply:
This part was completely missing in the initial impl, thanks very much for finding it!